static/app/components/modals/sudoModal.tsx

```diff
@@ -35,12 +35,37 @@ type Props = WithRouterProps &
 type State = {
   busy: boolean;
   error: boolean;
+  errorType: string;
+  isFirstStep: boolean;
+  superuserAccessCategory: string;
+  superuserReason: string;
 };
 
 class SudoModal extends React.Component<Props, State> {
   state: State = {
     error: false,
+    errorType: '',
     busy: false,
+    isFirstStep: true,
+    superuserAccessCategory: '',
+    superuserReason: '',
+  };
+
+  handleSubmit = async data => {
+    const {api, superuser} = this.props;
+
+    if (this.state.isFirstStep && superuser) {
+      this.setState({isFirstStep: false});
```

Comment:
This could all be the same `setState`
```
this.setState({
  isFirstStep: false,
  superuserAccessCategory: data.superuserAccessCategory,
  superuserReason: data.superuserReason,
})
```